Rakefile

```diff
@@ -47,8 +55,12 @@ namespace :test do
       Dir.chdir(work_dir) { system "bundle exec rackup #{rackup_opts}" }
     end
 
+    task :wait do
+      5.times { if need_to_wait?(runner_uri) then sleep(1) else break end }
```

Comment:
Hm, can't we actually put the retry logic in `need_to_wait`? See [retry](http://ruby-doc.com/docs/ProgrammingRuby/html/tut_expressions.html#UM).